### PR TITLE
Remove activation predicates

### LINKED ISSUE #52: Remove fork activation check functions
Remove activation check code e.g. `IsUAHFEnabled`

### PR DESCRIPTION
**Pull request by EyeOfPython**
Closes #52.

- Removes `IsUAHFenabled`, `IsDAAEnabled`, `IsMagneticAnomalyEnabled`, `IsGravitonEnabled`, `IsPhononEnabled` and `IsAxionEnabled`, but keeps the empty file in place for future upgrades.
- Removes all the block heights for past Bitcoin and Bitcoin Cash upgrades.
- Removes all the MTP timestamps for Bitcoin Cash upgrades.
- Removes the tests for the activation predicates, but keeps the empty file in place for future upgrades.

### CODE REVIEW

**Review by schancel — APPROVED** [review]

**Comment by schancel** [comment]
We have some conflicts

**Comment by EyeOfPython** [comment]
Will fix.

File: src/validation.cpp [hunk 1]
```diff
@@ -278,7 +278,7 @@ bool CheckSequenceLocks(const CTxMemPool &pool, const CTransaction &tx,
 static bool IsReplayProtectionEnabled(const Consensus::Params &params,
                                       int64_t nMedianTimePast) {
     return nMedianTimePast >= gArgs.GetArg("-replayprotectionactivationtime",
```
Comment (schancel) [inline src/validation.cpp:280]: We should probably just fork along with ABC's timing.

**Comment by schancel** [comment]
LGTM

**Pull request merged**